cli/send.go

```diff
@@ -244,3 +255,107 @@ var SendCmd = &cli.Command{
 		return nil
 	},
 }
+
+func handleCSVSend(cctx *cli.Context, csvFile string) error {
+	srv, err := GetFullNodeServices(cctx)
+	if err != nil {
+		return err
+	}
+	defer srv.Close() //nolint:errcheck
+
+	ctx := ReqContext(cctx)
+
+	// Add logic to handle --from address
+	var fromAddr address.Address
+	if from := cctx.String("from"); from != "" {
+		addr, err := address.NewFromString(from)
+		if err != nil {
+			return err
+		}
+		fromAddr = addr
+	} else {
+		defaddr, err := srv.FullNodeAPI().WalletDefaultAddress(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get default address: %w", err)
+		}
+		fromAddr = defaddr
+	}
+
+	// Print sending address
+	_, _ = fmt.Fprintf(cctx.App.Writer, "Sending messages from: %s\n", fromAddr.String())
+
+	fileReader, err := os.Open(csvFile)
+	if err != nil {
+		return xerrors.Errorf("read csv: %w", err)
+	}
+	defer func() {
+		if err := fileReader.Close(); err != nil {
+			log.Errorf("failed to close csv file: %v", err)
+		}
+	}()
+
+	r := csv.NewReader(fileReader)
+	records, err := r.ReadAll()
+	if err != nil {
+		return xerrors.Errorf("read csv: %w", err)
+	}
+
+	// Validate header
+	if len(records) == 0 ||
```

Comment:
```suggestion
	if len(records) == 0 ||
		len(records[0]) != 4 ||
```

probably also should `len(e) != 4` check each of the records in the loop below instead of assuming there's 4 in each one